### PR TITLE
cluster up: prevent admin templates from instantiating on older versions

### PR DESCRIPTION
**Pull request by csrwng**
Fixes https://github.com/openshift/origin/issues/14357

### CODE REVIEW

**Comment by csrwng** [comment]
@openshift/devex ptal
/cc @jcantrill @spadgett

File: pkg/bootstrap/docker/up.go [hunk 1]
```diff
@@ -917,7 +919,18 @@ func (c *ClientStartConfig) ImportTemplates(out io.Writer) error {
...
+		return err
+	}
+	if shouldImportAdminTemplates(version) {
+		return c.importObjects(out, "kube-system", adminTemplateLocations)
+	}
+	return nil
+}
+
+func shouldImportAdminTemplates(v semver.Version) bool {
+	return v.GT(openshiftVersion35)
```
Comment (bparees) [inline pkg/bootstrap/docker/up.go]: this means we'll install them on 3.5.1, right? is that ok?

Comment (csrwng) [inline pkg/bootstrap/docker/up.go]: you're right ... no, it needs to be 3.6 ... will change the condition to >= 3.6.0

Comment (csrwng) [inline pkg/bootstrap/docker/up.go]: fixed

**Comment by bparees** [comment]
lgtm

**Comment by csrwng** [comment]
@bparees I fixed the error check in the ServerVersion method, it was backwards ... should cache the result if there is no error. If it still looks good to you is it ok to merge?

**Comment by bparees** [comment]
ah, i thought that was intentional that you were clearing part of the version if you got an error.  so much for my faith in you :)

**Comment by bparees** [comment]
[merge][severity:bug]

**Comment by csrwng** [comment]
Flake https://github.com/openshift/origin/issues/13980

**Comment by smarterclayton** [comment]
Why does template creation fail the cluster up?  We should just warn ("couldn't add template X: reason")

**Comment by smarterclayton** [comment]
This isn't really versioning - this is that people are writing templates that aren't compatible on older versions because of api groups.  That's a much more fundamental problem.

**Comment by csrwng** [comment]
@smarterclayton the problem with simply warning about the template creation error is that we'll have a partially imported template... is that ok?

**Pull request merged**